ecleankernel/__main__.py

```diff
@@ -256,6 +264,40 @@ def main(argv: typing.List[str]) -> int:
                 bootloader=bootloader,
                 destructive=args.destructive)
 
+            if hook_dirs:
+                preserved_kernels = [k for k in kernels
+                                     if k not in removals]
+
+                def as_version_list(kernels: typing.Iterable[Kernel]) -> str:
+                    versions = [x.version for x in kernels]
+                    return " ".join(versions)
```

Comment:
Inline it in `.join()` as iteration, you don't need to create a temporary list for this.